4/README.md

```diff
@@ -0,0 +1,346 @@
+status: draft
+editor: [Sergi Blanchi-Torné](mailto:)
+contributors: [Michal Liszcz](mailto:michal.liszcz@s2innovation.com)
+---
+
+This document describes the Tango Attribute model specification version 5.0.
+
+
+## Preamble
+
+Copyright (c) 2019 Tango Controls
+
+This Specification is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 3 of the License, or (at your option) any later version. This Specification is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with this program; if not, see <http://www.gnu.org/licenses>.
+
+This Specification is a [free and open standard](http://www.digistan.org/open-standard:definition) and is governed by the Digital Standards Organization's [Consensus-Oriented Specification System](http://www.digistan.org/spec:1/COSS).
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+
+## Tango Attribute specification
+
+This specification is intended to formally document the Tango Attribute static
+model. Runtime representation of Attribute in conforming Tango implementations
+MUST follow this specification.
+
+
+### Goals
+
+An Attribute is a Tango concept representing a physical quantity of a device.
+The main purpose of an Attribute is to provide read and (optionally) write
+access to this quantity.
+
+In object oriented terminology, the Attribute corresponds to an instance
+variable (also called a field or a member) of a Device object.
+See [RFC 2/Device](/2) for the definition of the Device.
+
+### Use Cases
+
+Some example use cases of an Attribute are:
+* an Attribute can represent a position of a motor device,
+* an Attribute can represent a temperature of a thermocouple device.
+
+
+## Specification
+
+An Attribute has:
+* A set of static metadata that constitute to Attribute's definition,
+* A set of dynamically configurable properties, e.g. for alarming purposes,
+* A set of runtime parameters describing Attribute's value at given point
+  in time.
+
+### Attribute definition
+
+The static metadata is part of the Attribute definition. It MUST be defined
+before an Attribute is created by any Tango implementation and MUST NOT change
+at runtime after the Attribute is initialized.
+
+An Attribute MUST have associated following static metadata:
+* *name*, a string identifying the Attribute.
+  It MUST be unique [^1] among all Attributes of a particular device.
+  It MAY contain any character (including multi-byte characters)
+  except that it MUST NOT contain `\0` character.
```

Comment:
You can't have \0 (since it's end of string), but you can have  other bytes like \1, \2, ... But - such names will not be accepted by Pogo (which expects only ASCII letters and numbers).

I can remove the note about \0 ("except that it MUST NOT contain `\0` character.")

Do we want to remove the "any character" part as well? - "It MAY contain any character (including multi-byte characters)". Even though it will not work with Pogo, I think non-English letters in attribute names are still a valid use case.

Reply:
> You can't have \0 (since it's end of string), but you can have other bytes like \1, \2, ... But - such names will not be accepted by Pogo (which expects only ASCII letters and numbers).
> 
> I can remove the note about \0 ("except that it MUST NOT contain `\0` character.")

I don't know what would be the best. When reading it I just found it confusing:
> *name*, a **string** identifying the Attribute.
> it MUST NOT contain `\0` character.

A C string contains a `\0` at the end by definition so this seemed a bit contradictory to me.
 
> Do we want to remove the "any character" part as well? - "It MAY contain any character (including multi-byte characters)". Even though it will not work with Pogo, I think non-English letters in attribute names are still a valid use case.

Non-English letters/multi-byte characters in attribute names can be a valid use case indeed.  There might be some precautions needed for this to work correctly though. I think character set encoding configuration is important in this case on the database side , on the CORBA side and/or on the OS side.
I don't know whether some users are already doing it. This use case is not tested in cppTango I think.

Reply:
So I've removed the confusing note about the `\0`. I've left the information about multi-byte characters.
* Tango should work (with minor tweaking, see [TangoDatabase#16](https://github.com/tango-controls/TangoDatabase/issues/16)) with any Database chacter set.
* According to the answers in [this stackoverflow question](https://stackoverflow.com/questions/402283/stdwstring-vs-stdstring), `std::string` can hold unicode.
* Simple test shows it is possible to read attribute with non-ASCII name.